### ISSUE TITLE
Build url not set on commit status

### ISSUE THREAD

**Issue by marcokrikke** [jenkinsci/ghprb-plugin#206]
Jenkins version: 1.631
GHPRB version: 1.29

When a build starts or completes, the Job url is not set on the GitHub status:

![screen shot 2015-10-01 at 11 26 45](https://cloud.githubusercontent.com/assets/1352315/10217322/5957d3b6-682f-11e5-9004-87b138492e88.png)

I do not have any custom settings for the status url:

![screen shot 2015-10-01 at 11 24 16](https://cloud.githubusercontent.com/assets/1352315/10217330/62ddb31a-682f-11e5-9856-2a7b2f467074.png)

In the build log, it shows:

````
Setting status of xxx to PENDING with url  and message: 'Build started.'
[..]
Setting status of xxx to FAILURE with url  and message: ' No test results found.'
````

As can be seen, the url variable is not set.

This issue seems to be related to the changes in ddcf9788fe056d6f997121900189857b4db3d695 in `GhprbSimpleStatus`. In the `createCommitStatus` method:

````java
String url = envVars.get("BUILD_URL");
if (StringUtils.isEmpty(url)) {
    url = envVars.get("JOB_URL");
}

if ("--none--".equals(statusUrl) || statusUrl == null) {
    url = "";
} else if (!StringUtils.isEmpty(statusUrl)) {
    url = Ghprb.replaceMacros(build,  listener, statusUrl);
}
````

First, the url gets set to the Build url (which I would expect to show in the GitHub status). But since the `statusUrl` variable is `null`, the url then gets set to an empty string.

Any workaround available for this?

**Comment by mmitche** [comment]
I've seen the same behavior, but oddly enough it only repros when jobs are within a folder.

**Comment by mmitche** [comment]
This is a bug combination of a bug in the job-dsl-plugin and the change @marcokrikke mentioned above.  The xml element statusUrl is omitted from the generated xml, which causes url to be set to "".  If you were to open and re-save a generated job, the statusUrl element gets generated and the plugin works again.

**Comment by marcokrikke** [comment]
Thanks for looking into this Matt, and good guess we are using the job-dsl-plugin. I have checked the config.xml before and after re-saving the job, and indeed the statusUrl element has been added afterwards.

I hope the job-dsl-plugin gets fixed soon, and will use a custom build for now.

**Comment by marcokrikke** [comment]
This is fixed in job-dsl-plugin 1.40.

Thanks again for the PR @mmitche.
